jujucrashdump/crashdump.py

```diff
@@ -176,12 +189,28 @@ def __init__(self, model, max_size, extra_dirs, output_dir=None,
         self.timeout = timeout
         self.journalctl = journalctl
 
+    def get_all(self):
+        machines = {}
+        juju_status = yaml.load(open('juju_status.yaml', 'r'),
+                                Loader=yaml.FullLoader)
+        for machine, machine_data in juju_status['machines'].iteritems():
+            machines[machine] = machine_data['ip-addresses']
+            if 'containers' in juju_status['machines'][machine]:
+                for container, container_data in \
+                  juju_status['machines'][machine]['containers'].iteritems():
+                    machines[container] = container_data['ip-addresses']
+        return machines
+
     def _run_all(self, cmd):
-        run_cmd("timeout %ds juju run --all -- sh -c '%s'" % (
-            self.timeout, cmd))
+        all_machines = self.get_all()
+        ssh_cmd = 'ssh -o StrictHostKeyChecking=no'
+        with concurrent.futures.ThreadPoolExecutor(max_workers=10) as executor:
+            [executor.submit(run_ssh, ips, self.timeout, ssh_cmd, cmd)
```

Comment:
Do we need to await these threads to ensure they complete successfully or that the main thread doesn't terminate before they're complete?

Also, `multiprocessing` is used elsewhere, but since the work is almost entirely I/O bound anyway, maybe it would be more efficient (or at least consistent) to convert everything to use the `ThreadPoolExecutor`.

Reply:
I am not 100% sure, but this is what doc says
<>
shutdown(wait=True)
Signal the executor that it should free any resources that it is using when the currently pending futures are done executing....
You can avoid having to call this method explicitly if you use the with statement, which will shutdown the Executor (waiting as if Executor.shutdown() were called with wait set to True)
</>

converting multiprocessing to threadpool is probably a good idea, but not for this PR...